lib/ex_webrtc/peer_connection.ex

```diff
@@ -690,28 +772,35 @@ defmodule ExWebRTC.PeerConnection do
       case find_transceiver(transceivers, mid) do
         {idx, %RTPTransceiver{} = tr} ->
           new_tr = RTPTransceiver.update(tr, mline, state.config)
-          notify_if_new_track(mline, new_tr, state.owner)
-
+          new_tr = process_remote_track(new_tr, mline, state.owner)
           List.replace_at(transceivers, idx, new_tr)
 
         nil ->
           new_tr = RTPTransceiver.from_mline(mline, state.config)
-          notify_if_new_track(mline, new_tr, state.owner)
+          new_tr = process_remote_track(new_tr, mline, state.owner)
           transceivers ++ [new_tr]
       end
     end)
   end
 
-  defp notify_if_new_track(mline, tr, owner) do
-    # TODO: applying offers back to back (without answer)
-    # will result in multiple messages for the same track
-    # because current_direction is set in set_remote_description
+  # see W3C WebRTC 5.1.1
+  defp process_remote_track(transceiver, mline, owner) do
     direction = SDPUtils.get_media_direction(mline)
 
-    if direction in [:sendrecv, :sendonly] and
-         tr.current_direction not in [:sendrecv, :recvonly] do
-      notify(owner, {:track, tr.receiver.track})
+    cond do
+      direction in [:sendrecv, :recvonly] and
```

Comment:
I changed from `[:sendrecv, :sendonly]` to `[:sendrecv, :recvonly]` according to spec

Reply:
Remember that the W3C 5.1.1 assumes that `direction` is the remote m-line direction from this peer's point of view (it is swapped in 4.4.1.5.4.7.10.2.10, "10. Let `direction` be ...). We don't do the swap (afaik), that's why I used the `:sendonly`. If I understand correctly, now your trying to emit the track event for remote recvonly tracks (so sendonly from our peer's perspective).

Reply:
Good point, I didn't check this